utils/env_preferences/src/parse/posix.rs

```diff
@@ -0,0 +1,621 @@
+    /// (unknown modifiers will be silently ignored).
+    ///
+    /// If the "logging" feature is enabled, this ignored data will be logged as a warning.
+    pub fn try_convert_lossy(&self) -> Result<Locale, ConversionError> {
+        // Check if the language needs to be mapped to a BCP-47 language subtag
+        let language_code = POSIX_LANGUAGE_NAMES
+            .binary_search_by_key(&self.language, |(language, _code)| language)
+            // `binary_search_by_key()` guarantees a valid index
+            .map(|index| POSIX_LANGUAGE_NAMES[index].1);
+
+        let mut language = match language_code {
+            // Found a matching language code
+            Ok(language) => language,
+            // Parse the language directly
+            Err(_) => match Language::try_from_str(self.language) {
+                Ok(language) => language,
+                Err(_) => {
+                    return Err(ConversionError::InvalidLanguage {
+                        start_offset: 0,
+                        end_offset: self.language.len(),
+                    })
+                }
+            },
+        };
+
+        let region = if let Some(territory) = self.territory {
+            match Region::try_from_str(territory) {
+                Ok(region) => Some(region),
+                Err(_) => {
+                    return Err(ConversionError::InvalidRegion {
+                        // Add 1 to skip the delimiter
+                        start_offset: self.language.len() + 1,
+                        // Add 1 to include the final character
+                        end_offset: self.language.len() + territory.len() + 1,
+                    });
+                }
+            }
+        } else {
+            None
+        };
+
+        if let Some(codeset) = self.codeset {
+            icu_provider::log::warn!("Ignoring codeset: `{codeset}`");
+        }
+
+        let mut extensions = Extensions::new();
+        let mut script = None;
+        let mut variants = vec![variant!("posix")];
+
+        if let Some(modifier) = self.modifier {
+            match modifier.to_ascii_lowercase().as_str() {
+                "euro" => {
+                    extensions.unicode.keywords.set(key!("cu"), value!("eur"));
+                }
+                // Known script modifiers
+                "cyrillic" => script = Some(script!("Cyrl")),
+                "devanagari" => script = Some(script!("Deva")),
+                "latin" => script = Some(script!("Latn")),
+                // Saaho seems to be the only "legacy variant" that appears as a modifier:
+                // https://www.unicode.org/reports/tr35/#table-legacy-variant-mappings
+                "saaho" => {
+                    // The only known locale with this modifier is `aa_ER`
+                    if self.language != "aa" || self.territory != Some("ER") {
+                        icu_provider::log::warn!("Overriding language: `{}`->`ssy`", self.language);
+                    }
```

Comment:
`question`: the LDML you referenced indicates that `saaho` should result in language being set to `ssy`, irrelevant of the original `language`. value, why do you narrow it to `aa`?

Reply:
Also, similar concern about logging as above.